sql/clients_last28_v1.sql

```diff
@@ -0,0 +1,42 @@
+WITH
+  current_sample AS (
+  SELECT
+    * EXCEPT (submission_date_s3),
+    0 AS days_since_seen,
+    -- Active User means a client who has total_daily_uri >= 5
+    -- https://docs.telemetry.mozilla.org/cookbooks/active_dau.html
+    IF(scalar_parent_browser_engagement_total_uri_count_sum >= 5,
+      0,
+      NULL) AS days_since_active_user
+  FROM
+    clients_daily_v6
+  WHERE
+    submission_date_s3 = @submission_date ),
+  previous AS (
+  SELECT
+    * EXCEPT (submission_date) REPLACE(
+      -- omit values outside 28 day window
+      IF(days_since_active_user < 27,
+        days_since_active_user,
+        NULL) AS days_since_active_user)
+  FROM
+    clients_last28_v1
+  WHERE
+    submission_date = DATE_SUB(@submission_date, INTERVAL 1 DAY)
+    AND clients_last28_v1.days_since_seen < 27 )
+SELECT
+  @submission_date AS submission_date,
+  IF(current_sample.client_id IS NOT NULL,
+    current_sample,
+    previous).* EXCEPT (days_since_seen,
+      days_since_active_user),
+  COALESCE(current_sample.days_since_seen,
+    previous.days_since_seen + 1) AS days_since_seen,
+  COALESCE(current_sample.days_since_active_user,
+    previous.days_since_active_user + 1) AS days_since_active_user
+FROM
+  current_sample
```

Comment:
It's been irking me a bit that the names of these CTEs aren't more symmetrical: `previous` vs. `current_sample`. I think they could also stand to be shorter since they're repeated so often in the query. What do you think about prev and curr? prevday and currday?

Reply:
I like `prev` and `curr`, and I had the same thought (it would be `current`, but that's a keyword so i avoided it)